src/NuGet.Core/NuGet.Configuration/Settings/Settings.cs

```diff
@@ -246,7 +246,7 @@ bool useTestingGlobalPath
                 }
                 else
                 {
-                    var userSettingsDir = NuGetEnvironment.GetFolderPath(NuGetFolderPath.UserSettingsDirectory);
+                    var userSettingsDir = NuGetEnvironment.GetFolderPath(NuGetFolderPath.UserSettingsDirectory) ?? String.Empty;
```

Comment:
@zhili1208 can you please take over this. I think the right fix here is not to use string.Empty but instead return no user specific settings from appdata like the code below

```c#
             if (appDataSettings != null)
             {		             {
                 validSettingFiles.Add(appDataSettings);		                 validSettingFiles.Add(appDataSettings);
             }		             }
```